### PR TITLE
Implement storage policy configuration lint and query tool

### PR DESCRIPTION
**Pull request by NicolasT**
Lint valid file:
```
$ swift-scality-backend storage-policy-lint --config scality-storage-policy.ini
$ echo $?
0
```

Query valid file:
```
$ swift-scality-backend storage-policy-query --config scality-storage-policy.ini read 1 sfo paris
Query
-----
Policy index: 1
Action: read
Location hints: ['sfo', 'paris']

Result
------
0: ['http://sfo1.int/arc6+3']
1: ['http://paris1.int/arc6+3', 'http://paris2.int/arc6+3']
```

Lint invalid file:
```
$ echo '[ring:invalid]' | swift-scality-backend storage-policy-lint -c -
Configuration error: Section 'ring:invalid' lacks a 'location' setting
$ echo $?
1
```

Default configuration file is `SWIFT_DIR/scality-storage-policies.ini`.

### CODE REVIEW

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/swift-tox/137/<h2>Build result: FAILURE</span></h2>[...truncated 23 lines...]Triggering swift-tox » py27-swift2.2.0,centos7-largeTriggering swift-tox » py27-swift1.13.1,ubuntu12-largeTriggering swift-tox » py26-swift1.13.1,centos6-largeTriggering swift-tox » py26-swifthead,centos6-largeTriggering swift-tox » pep8,ubuntu14-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-large-434Configuration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-434 is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-largeswift-tox » py27-swifthead,ubuntu12-large completed with result SUCCESSswift-tox » py27-swift2.2.0,ubuntu14-large completed with result UNSTABLEswift-tox » py26-swift2.2.0,centos6-large completed with result UNSTABLEswift-tox » py27-swift1.13.1,centos7-large completed with result UNSTABLEswift-tox » py27-swifthead,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift1.13.1,ubuntu14-large completed with result SUCCESSswift-tox » py27-swift2.2.0,ubuntu12-large completed with result UNSTABLEswift-tox » py27-swifthead,centos7-large completed with result UNSTABLEswift-tox » py27-swift2.2.0,centos7-large completed with result SUCCESSswift-tox » py27-swift1.13.1,ubuntu12-large completed with result FAILUREswift-tox » py26-swift1.13.1,centos6-large completed with result UNSTABLEswift-tox » py26-swifthead,centos6-large completed with result UNSTABLEswift-tox » pep8,ubuntu14-large completed with result SUCCESSSetting status of 49512f4d09dfe0280cc8c9e4e23eec5a8bebf0ba to FAILURE with url https://37.187.159.67:5443/job/swift-tox/137/ and message: Merged build finished.
Test FAILed.

File: swift_scality_backend/cli.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''Command-line utilities.'''
+
+import argparse
```
Comment (JordanP) [inline swift_scality_backend/cli.py]: No argparse on Py2.6. You may want to print a warning that this tool doesn't support Python 2.6

File: test/unit/test_cli.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    def test_lint_fails_on_malformed_file(self):
+        config = 'test'
+
+        args = Namespace(
+            config=StringIO(config))
+
+        with FakeStream(sys, 'stderr') as stderr:
+            rc = cli.storage_policy_lint(args)
+
+            self.assertRegexpMatches(
```
Comment (JordanP) [inline test/unit/test_cli.py]: I think you should use utils.assertRegexpMatches here for Python 2.6 compatibility. Or probably skip these tests on Python 2.6.

File: swift_scality_backend/cli.py (hunk 1 above)
Comment (NicolasT) [inline swift_scality_backend/cli.py]: OK.

File: test/unit/test_cli.py (hunk 2 above)
Comment (NicolasT) [inline test/unit/test_cli.py]: OK.

File: swift_scality_backend/cli.py (hunk 1 above)
Comment (NicolasT) [inline swift_scality_backend/cli.py]: Actually, turns out the `argparse` package is installed as a transient dependency when using Python 2.6 (but maybe only through `test-requirements.txt`). Will add work-around code.

**Comment by scality** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://37.187.159.67:5443/job/swift-tox/142/<h2>Build result: FAILURE</span></h2>[...truncated 27 lines...]Triggering swift-tox » pep8,ubuntu14-largeConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-1e0 is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executorConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: All nodes of label ‘ubuntu12-large’ are offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: ubuntu12-large-21a is offlineConfiguration swift-tox » py27-swifthead,ubuntu12-large is still in the queue: Waiting for next available executor on ubuntu12-large-21aswift-tox » py27-swifthead,ubuntu12-large completed with result FAILUREswift-tox » py27-swift2.2.0,ubuntu14-large completed with result FAILUREswift-tox » py26-swift2.2.0,centos6-large completed with result FAILUREConfiguration swift-tox » py27-swift1.13.1,centos7-large is still in the queue: All nodes of label ‘centos7-large’ are offlineConfiguration swift-tox » py27-swift1.13.1,centos7-large is still in the queue: Waiting for next available executorConfiguration swift-tox » py27-swift1.13.1,centos7-large is still in the queue: centos7-large-84a is offlineswift-tox » py27-swift1.13.1,centos7-large completed with result FAILUREswift-tox » py27-swifthead,ubuntu14-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu14-large completed with result FAILUREswift-tox » py27-swift2.2.0,ubuntu12-large completed with result FAILUREswift-tox » py27-swifthead,centos7-large completed with result FAILUREswift-tox » py27-swift2.2.0,centos7-large completed with result FAILUREswift-tox » py27-swift1.13.1,ubuntu12-large completed with result FAILUREswift-tox » py26-swift1.13.1,centos6-large completed with result FAILUREswift-tox » py26-swifthead,centos6-large completed with result FAILUREswift-tox » pep8,ubuntu14-large completed with result FAILURESetting status of 3ea445f2782c01d453df1d3b26e7e11b3c11bb0c to FAILURE with url https://37.187.159.67:5443/job/swift-tox/142/ and message: Merged build finished.
Test FAILed.

**Comment by JordanP** [comment]
https://37.187.159.67:5443/view/Swift/job/swift-tox/145/

LGTM

**Pull request merged**